### PR TITLE
Replace Shader GL calls with Program calls where they were invalid

### PR DESCRIPTION
**Pull request by eigenraven**
GL linked programs use separate functions from shaders, the original code seems to work on some drivers by accident.

Closes <https://github.com/GTNewHorizons/lwjgl3ify/issues/1>

### CODE REVIEW

**Comment by ReikaKalseki** [comment]
If these work only on "some drivers" why are you the first to report this, more than three years after the shaders were added?

**Comment by eigenraven** [comment]
Because it's error handling code, and it doesn't seem you ship versions with shader errors. If you deliberately add a linking error into your shader (e.g. unmatched attributes across the vertex and fragment shader), the current code would most likely not detect the error. Currently, no matter if the linking fails or succeeds, calling `glGetShaderi` with `GL_LINK_STATUS` will set the gl error and not update the implicit array lwjgl gives to the native `glGetShaderiv` function, which means lwjgl2 will pass a 1 to you (cached from the previous shader status check call). According to the specs (<https://registry.khronos.org/OpenGL-Refpages/gl2.1/xhtml/glGetShader.xml>), `GL_LINK_STATUS` is not a valid enum value for that function and it should set the error you can get with `glGetError` to `GL_INVALID_ENUM`. It could also be a `GL_INVALID_OPERATION` if the index of the program you pass in is not also an index of a shader, and the index allocation strategy depends on the specific driver implementation.

In the linked issue a player, and later also me when reproducing the issue, ran into this code being broken because in my lwjgl2->3 compatibility shim a local array initialized with 0 is used for interactions with the native code instead of a global intbuffer containing 1 from the previous (valid) call. This triggered the `if` statement checking the program status to throw an exception in DragonAPI on load. There are ways I could work around this in my shim layer, but because you still maintain your mods I assumed you would want to fix these callsites.

In short, in a typical minecraft installation the bug is hidden due to a combination of lwjgl2 internal caching and no failure in your shaders, my lwjgl upgrade shim made it visible.

**Comment by Abalieno** [comment]
...For (unrequested) context: Raven has made possible to run most of 1.7.10 mods on Java 17/19 and lwjgl3, so that's the reason why this appears just now.

You (Reika) might not have any interest on this other possibility/compatibility, but since it was proven to be a bug it could be easily fixed without any other negative repercussion...

(but I wonder if FalsePattern reported it just to make fun of it...)

**Comment by ReikaKalseki** [comment]
I get a lot of people coming and claiming old code was "always broken and nonfunctional", when their definition of "nonfunctional" is "breaks under very specific circumstances that are not typically worth accounting for", hence the question.

While this half-qualifies, because there *is* some value in fixing it due to wanting the error handling to work universally, and because - as stated above - it will not actually affect "desired" code, it has been implemented, if slightly differently.
